### PR TITLE
Add docusaurus

### PR DESCRIPTION
**Pull request by Dr-Electron**
I tried to add Docusaurus without changing too much stuff. I think this is the most minimal setup I can come up with 🤔 

Some stuff I had to change and which maybe needs to be discussed:

- [x] Had to add an introduction page in the `tips` folder which imports the `README`
- [x] Had to change `<br>` to `</br>` as Docusaurus uses XHTML
- [x] Added some missing html tags
- [x] Had to escape `<byte>`. But I saw in some TIPs ByteArrays was used. So I guess we could use that too 🤔 
- [x] Wrote a custom sidebar generator so that we get `{{TIP-Number}}-{{TIP-Name}}` as label. This can of course be changed

You can test the branch by running:
(If you don't already have it enabled)
```
corepack enable
```
And then
```
yarn && yarn start
```

TODO:

- [x] Add broken links detection
- [x] Test importing into wiki

### CODE REVIEW

**Comment by jlvandenhout** [comment]
If I can move and adapt the CLI this week, we can get rid of the Yarn stuff and just have the Docusaurus config, that would make it even more minimal. Would that be soon enough or do we want to merge this ASAP?

File: tips/TIP-0029/tip-0029.md [hunk 1]
```diff
@@ -188,6 +188,8 @@ The following table describes the entirety of a _Milestone Payload_ in its seria
   </tr>
 </table>
 
+¹: [TIP-4](../TIP-0004/tip-0004.md)
```
Comment (jlvandenhout) [inline tips/TIP-0029/tip-0029.md:191]:
```suggestion
¹: See [TIP-4](../TIP-0004/tip-0004.md).
```

Just to keep the original text.

**Review by jlvandenhout — CHANGES_REQUESTED** [review]

**Review by robertf26 — APPROVED** [review]

**Review by lucas-tortora — APPROVED** [review]

File: tips/TIP-0029/tip-0029.md [hunk 2]
```diff
@@ -260,8 +263,8 @@ Syntactic validation can be checked from the _Milestone Essence_ plus the blocks
...
     - `Index Number` must increment by `1`. 
     - `Timestamp` must be strictly larger (i.e. at least one second later).
-    - `Inlcusion Merkle Root` must match the Merkle tree hash of the IDs of all blocks in _White Flag Ordering_ (as described in <a href="../TIP-0002/tip-0002.md#deterministically-ordering-the-tangle">TIP-2</a>) that are newly referenced. (This always includes at least one valid milestone block with `Previous Milestone ID`.)
-    - `Applied Merkle Root` must match the Merkle tree hash of the not-ignored state-mutating transactions that are newly referenced (see <a href="../TIP-0002/tip-0002.md">TIP-2</a>).
+    - `Inlcusion Merkle Root` must match the Merkle tree hash of the IDs of all blocks in _White Flag Ordering_ (as described in [TIP-2](../TIP-0002/tip-0002.md#deterministically-ordering-the-tangle)) that are newly referenced. (This always includes at least one valid milestone block with `Previous Milestone ID`.)
```
[the remark this answers is not in the document]
Comment (Dr-Electron) [reply tips/TIP-0029/tip-0029.md:266]: The only reason for absolute links was that this pages aren’t part of the docs we can/will add to the wiki.
But here TIP-2 will be part of the wiki so you can switch to that page directly on the wiki without leaving it

**Comment by Dr-Electron** [comment]
You can look at a wiki preview here:
https://iota-wiki-it-iota-iota1.vercel.app/tips

**Comment by Dr-Electron** [comment]
Sorry still new to vercel 😅 .
This one should work tho: https://iota-wiki-9auyou25l-iota1.vercel.app/tips/

File: CONTRIBUTE.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+> **Note**
+> This guide is meant for local development and contributing to the repository only, for guidance on the TIPs process and how to propose Tangle improvements, please see the [README](README.md)
+
+
+# Documentation
+
+The TIPs online documentation is built using [Docusaurus 2](https://docusaurus.io/). The deployment is done through a centralized build from [IOTA WIKI](https://github.com/iota-community/iota-wiki). To run a local instance the [IOTA WIKI CLI](https://github.com/iota-community/iota-wiki-cli) is used.
```
Comment (jlvandenhout) [inline CONTRIBUTE.md:7]:
```suggestion
The TIPs online documentation is built using [Docusaurus 2](https://docusaurus.io/). The deployment is done through a centralized build from [IOTA WIKI](https://github.com/iota-wiki/iota-wiki). To run a local instance, the [IOTA WIKI CLI](https://www.npmjs.com/package/@iota-wiki/cli) is used.
```

**Review by jlvandenhout — CHANGES_REQUESTED** [review]

File: CONTRIBUTE.md (hunk 3 above)
Comment (Dr-Electron) [reply CONTRIBUTE.md:7]: Could you fix this in our template too? (We now need to update it anyway 😅)

**Review by jlvandenhout — APPROVED** [review]

**Review by lzpap — APPROVED** [review]
Amazing job with integrating the TIP repo content into the wiki! 👏 🚀 
Thanks for the effort @Dr-Electron @jlvandenhout @robertf26 and @lucas-tortora !

**Pull request merged**